browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -46,62 +46,19 @@ void SetAdblockCnameHostResolverForTesting(
   g_testing_host_resolver = host_resolver;
 }
 
-void ShouldBlockAdOnTaskRunner(std::shared_ptr<BraveRequestInfo> ctx,
-                               base::Optional<std::string> canonical_name) {
-  bool did_match_rule = false;
-  bool did_match_exception = false;
-  bool did_match_important = false;
-  if (!ctx->initiator_url.is_valid()) {
-    return;
-  }
-  std::string source_host = ctx->initiator_url.host();
-
-  g_brave_browser_process->ad_block_service()->ShouldStartRequest(
-      ctx->request_url, ctx->resource_type, source_host, &did_match_rule,
-      &did_match_exception, &did_match_important, &ctx->mock_data_url);
-  if (did_match_important) {
-    ctx->blocked_by = kAdBlocked;
-    return;
-  }
-
-  if (canonical_name.has_value() &&
-      ctx->request_url.host() != *canonical_name && *canonical_name != "") {
-    GURL::Replacements replacements = GURL::Replacements();
-    replacements.SetHost(
-        canonical_name->c_str(),
-        url::Component(0, static_cast<int>(canonical_name->length())));
-    const GURL canonical_url = ctx->request_url.ReplaceComponents(replacements);
-
-    g_brave_browser_process->ad_block_service()->ShouldStartRequest(
-        canonical_url, ctx->resource_type, source_host, &did_match_rule,
-        &did_match_exception, &did_match_important, &ctx->mock_data_url);
-  }
-
-  if (did_match_important || (did_match_rule && !did_match_exception)) {
-    ctx->blocked_by = kAdBlocked;
-  }
-}
-
-void OnShouldBlockAdResult(const ResponseCallback& next_callback,
-                           std::shared_ptr<BraveRequestInfo> ctx) {
-  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
-  if (ctx->blocked_by == kAdBlocked) {
-    brave_shields::BraveShieldsWebContentsObserver::DispatchBlockedEvent(
-        ctx->request_url, ctx->frame_tree_node_id, brave_shields::kAds);
-  }
-  next_callback.Run();
-}
-
-void ShouldBlockAdWithOptionalCname(
-    scoped_refptr<base::SequencedTaskRunner> task_runner,
-    const ResponseCallback& next_callback,
-    std::shared_ptr<BraveRequestInfo> ctx,
-    const base::Optional<std::string> cname) {
-  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
-  task_runner->PostTaskAndReply(
-      FROM_HERE, base::BindOnce(&ShouldBlockAdOnTaskRunner, ctx, cname),
-      base::BindOnce(&OnShouldBlockAdResult, next_callback, ctx));
-}
+// Used to keep track of state between a primary adblock engine query and one
+// after CNAME uncloaking the request.
+typedef struct EngineFlags {
```

Comment:
no need for typedef in c++

Reply:
also I guess you have initialize fields with false

Reply:
removed the `typedef` and added `= false` initializers to the fields